### PR TITLE
Fixed mongo flapiness

### PR DESCRIPTION
**Pull request by bizob2828**
## Details
We were getting random failures seemingly only on main.  I was able to reproduce but very unclear root cause.  I suspect it is because in db.tap.js we were dropping the database and it was the same for cjs and esm.  I updated esm to use a different database but also took the time to centralize both collections and databases in common.{js,cjs} so make this easier to maintain going forward.

### CODE REVIEW

File: test/versioned/mongodb-esm/common.cjs [hunk 1]
```diff
@@ -13,7 +13,7 @@ const urltils = require('../../../lib/util/urltils')
 
 const MONGO_SEGMENT_RE = /^Datastore\/.*?\/MongoDB/
 const TRANSACTION_NAME = 'mongo test'
-const DB_NAME = 'integration'
+const DB_NAME = 'esmIntegration'
```
Comment (bizob2828) [inline test/versioned/mongodb-esm/common.cjs:16]: this is suspect is the actual fix. for esm tests connect to a diff database so when in db.tap.js we drop the database it won't cause random issues for the other test suite

**Comment by mrickard** [comment]
Index-based array access seems kind of fragile. Would there be an advantage to making the collections an object, with named properties for access?

**Comment by bizob2828** [comment]
> Index-based array access seems kind of fragile. Would there be an advantage to making the collections an object, with named properties for access?

yea I can see how it could be a little fragile but it's an improvement of having the same string littered. also that really wasn't the point of this PR. it was to fix the db name being shared across cjs and esm. Personally I don't feel like updating my regex to replace all with an object. how would the object look?

**Comment by mrickard** [comment]
@bizob2828 Understandable, and this is indeed an improvement!

My thought with an object would be `COLLECTIONS = { collection1, collection2 }` or some equivalent...but with only two collections to worry about, this is probably overkill.

**Comment by bizob2828** [comment]
> @bizob2828 Understandable, and this is indeed an improvement!
> 
> My thought with an object would be `COLLECTIONS = { collection1, collection2 }` or some equivalent...but with only two collections to worry about, this is probably overkill.

actually i could do this

**Review by mrickard — APPROVED** [review]

**Pull request merged**